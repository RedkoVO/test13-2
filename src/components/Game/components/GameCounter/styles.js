import WinnerLogo from '../../../../assets/images/winner.png'

export default theme => ({
  root: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'flex-start',
    justifyContent: 'space-between',
    height: '175px',
    color: '#fff',
    width: '100%'
  },
  counterHours: {
    width: '300px',
    height: '150px',
    marginRight: '20px',
    textAlign: 'center',
    backgroundColor: 'rgba(0,0,0,0.26)',
    borderRadius: '25px',

    [theme.breakpoints.down('md')]: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      padding: '0 5px',
      marginRight: '10px',
      minWidth: '95px',
      boxSizing: 'border-box'
    }
  },
  counterUsers: {
    width: '250px',
    height: '150px',
    marginRight: '20px',
    textAlign: 'center',
    backgroundColor: 'rgba(0,0,0,0.26)',
    borderRadius: '25px',

    [theme.breakpoints.down('md')]: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      padding: '0 5px',
      marginRight: '10px',
      minWidth: '95px',
      boxSizing: 'border-box'
    }
  },
  counterUsersVolume: {
    fontSize: '19px',
    fontWeight: 700,
    padding: 0,
    marginBottom: 0,

    [theme.breakpoints.down('md')]: {
      fontSize: '15px'
    }
  },
  counterContracts: {
    width: '600px',
    height: '150px',
    borderTopRightRadius: '0px',
    borderBottomRightRadius: '0px',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'flex-start',
    paddingLeft: '40px',
    boxSizing: 'border-box',
    textAlign: 'center',
    backgroundColor: 'rgba(0,0,0,0.26)',
    borderRadius: '25px',

    [theme.breakpoints.down('lg')]: {
      padding: '0 20px',
      minWidth: '258px'
    },
    [theme.breakpoints.down('md')]: {
      minWidth: '230px',
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      padding: '0 5px'
    }
  },
  counterWins: {
    width: '420px',
    height: '150px',
    borderTopLeftRadius: '0px',
    borderBottomLeftRadius: '0px',
    marginRight: '20px',
    background: `rgba(0,0,0,0.26)url(${WinnerLogo}) 0 0 no-repeat`,
    backgroundSize: '330px',
    backgroundPosition: '40px 10px',
    textAlign: 'center',
    backgroundColor: 'rgba(0,0,0,0.26)',
    borderRadius: '25px',

    [theme.breakpoints.down('md')]: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      padding: '0 5px',
      minWidth: '300px',
      backgroundSize: '280px',
      backgroundPosition: '10px 30px',
      marginRight: '10px'
    }
  },
  counterBiggestWin: {
    fontSize: '19px',
    fontWeight: 700,
    padding: 0,
    marginBottom: 0,

    [theme.breakpoints.down('md')]: {
      fontSize: '15px'
    }
  },
  counterHoursVolume: {
    fontSize: '19px',
    fontWeight: 700,
    padding: 0,
    marginBottom: 0,

    [theme.breakpoints.down('md')]: {
      fontSize: '15px'
    }
  },
  counterContractsVolume: {
    fontSize: '19px',
    fontWeight: 700,
    padding: 0,
    marginBottom: 0,

    [theme.breakpoints.down('md')]: {
      fontSize: '15px'
    }
  },
  counterContractsLink: {
    fontSize: '9px',
    color: '#fff',
    marginLeft: '10px'
  },
  counterValue: {
    fontSize: '34px',
    fontWeight: 700,

    [theme.breakpoints.down('md')]: {
      fontSize: '28px'
    }
  }
})
