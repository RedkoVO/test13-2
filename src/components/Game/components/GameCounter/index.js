import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import withStyles from '@material-ui/core/styles/withStyles'

import styles from './styles'

const GameCounter = ({ classes }) => (
  <div className={classes.root}>
    <div className={classes.counterWins}>
      <p className={classes.counterBiggestWin}>
        <span className={classes.counterValue}>4666.78 dcb</span> <br /> BIGGEST
        WIN <br /> FOR ALL TIME
      </p>
    </div>
    <div className={classes.counterHours}>
      <p className={classes.counterHoursVolume}>
        <span className={classes.counterValue}>167.12 dcb</span> <br /> VOLUME
        OF <br />
        24 HOURS
      </p>
    </div>
    <div className={classes.counterUsers}>
      <p className={classes.counterUsersVolume}>
        <span className={classes.counterValue}>785</span> <br /> USERS FOR <br />
        LAST 24 HOURS
      </p>
    </div>
    <div className={classes.counterContracts}>
      <p className={classes.counterContractsVolume}>
        <span className={classes.counterValue}>20243.12 dcb</span> <br /> ALL
        VOLUME OF <br /> SMART CONTRACT
      </p>
      <br />
      <Link to="/" className={classes.counterContractsLink}>
        0x7fdcd2a1e52f10c28cb7732f46393e297ecadda1
      </Link>
    </div>
  </div>
)

GameCounter.propTypes = {
  classes: PropTypes.object
}

export default withStyles(styles)(GameCounter)
