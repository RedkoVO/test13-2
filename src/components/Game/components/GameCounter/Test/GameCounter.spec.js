import * as React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'

import GameCounter from '../'

it('renders correctly', () => {
  const tree = renderer.create(
    <MemoryRouter keyLength={0}>
      <GameCounter />
    </MemoryRouter>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})