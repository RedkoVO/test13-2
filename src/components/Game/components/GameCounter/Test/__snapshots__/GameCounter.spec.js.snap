// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="GameCounter-root-1"
>
  <div
    className="GameCounter-counterWins-6"
  >
    <p
      className="GameCounter-counterBiggestWin-7"
    >
      <span
        className="GameCounter-counterValue-11"
      >
        4666.78 dcb
      </span>
       
      <br />
       BIGGEST WIN 
      <br />
       FOR ALL TIME
    </p>
  </div>
  <div
    className="GameCounter-counterHours-2"
  >
    <p
      className="GameCounter-counterHoursVolume-8"
    >
      <span
        className="GameCounter-counterValue-11"
      >
        167.12 dcb
      </span>
       
      <br />
       VOLUME OF 
      <br />
      24 HOURS
    </p>
  </div>
  <div
    className="GameCounter-counterUsers-3"
  >
    <p
      className="GameCounter-counterUsersVolume-4"
    >
      <span
        className="GameCounter-counterValue-11"
      >
        785
      </span>
       
      <br />
       USERS FOR 
      <br />
      LAST 24 HOURS
    </p>
  </div>
  <div
    className="GameCounter-counterContracts-5"
  >
    <p
      className="GameCounter-counterContractsVolume-9"
    >
      <span
        className="GameCounter-counterValue-11"
      >
        20243.12 dcb
      </span>
       
      <br />
       ALL VOLUME OF 
      <br />
       SMART CONTRACT
    </p>
    <br />
    <a
      className="GameCounter-counterContractsLink-10"
      href="/"
      onClick={[Function]}
    >
      0x7fdcd2a1e52f10c28cb7732f46393e297ecadda1
    </a>
  </div>
</div>
`;
